Assets/Mirror/Authenticators/TimeoutAuthenticator.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections;
+using UnityEngine;
+
+namespace Mirror.Authenticators
+{
+    /// <summary>
+    /// An authenticator that disconnects connections if they don't
+    /// authenticate within a specified time limit.
+    /// </summary>
+    [AddComponentMenu("Network/Authenticators/TimeoutAuthenticator")]
+    public class TimeoutAuthenticator : NetworkAuthenticator
+    {
+        public NetworkAuthenticator authenticator;
+
+        [Range(ushort.MinValue, ushort.MaxValue), Tooltip("Timeout to auto-disconnect in seconds. Set to 0 for no timeout.")]
+        public ushort timeout = 60;
```

Comment:
```suggestion
        public float timeout = 60;
```
Unity usually uses float or double for time. The 2 bytes that we save are hardly worth the inconvenience if someone wants 1.5s timeout etc.